ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -811,6 +813,9 @@ prop_diffusion_dns_can_recover defaultBearerInfo diffScript =
              in verify (Map.insert dns (addTime ttl' t) toRecover)
                         ttlMap'
                         recovered t evs
+        DiffusionLocalRootPeerTrace
+          (TraceLocalRootReconfigured _ _) ->
+            verify Map.empty ttlMap recovered t evs
```

Comment:
Yes and it is true, because when the configuration changes, we kill all the monitoring threads and restart them

Reply:
Not sure what your first question was then.

What happens is that we monitor our domain groups for resolving their DNS. And each domain name gets a monitoring thread.  These threads are only valid for the duration of the validity of out local configuration of root peers, see here: https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS.hs#L171 .

In our tests for this it was missing that when root peers are reconfigured we need to make sure our ttls are reset because that's what's going to happen in the actual implementation too.

Reply:
Thanks, I understand now.